dicomsort

```diff
@@ -60,26 +80,55 @@ fi
 for f in $files;
 do
 	echo $f
+	
+	#We are only using dicom_hdr to recognize dicom files now.  This seems clunky
+	#and could probably be sped up in the future.  All valid dicom files should have 
+	#the text DICM at offset 0x80 and we could just look for that in the future but 
+	#this is okay for now
 	dicom_hdr $f > $tempfile 
 	
 	# make sure it's a dicom file
-	if grep 'ERROR:' $tempfile 
-	then 
+	if grep 'ERROR:' $tempfile; then 
 	  continue
 	fi
 	
-	# I shortened the grep patterns and changed the awk delimiter to '//'
-	snum=`cat $tempfile | grep "Series Number" | awk -F '//' '{print $3}'`
-	anum=`cat $tempfile | grep "Acquisition Number" | awk -F '//' '{print $3}'`
-	inum=`cat $tempfile | grep "Instance Number" | awk -F '//' '{print $3}'`
+	# Now using dicom_hinfo to get specific tags instead of dicom_hdr and grep
+    snum=`dicom_hinfo -tag $snumtag -no_name $f`
+    anum=`dicom_hinfo -tag $anumtag -no_name $f`
```

Comment:
The subshell syntax that starts with the ` has been deprecated in favor of `$( your command here)`. The new `$()` makes quoting easier in certain situations